### PR TITLE
[ONNX] Add sanity check in CI for onnxbench

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #114907
* __->__ #110178

ONNX CI to run benchmark with `--quick` to validate the onnxbench infra.

### CODE REVIEW

File: scripts/onnx/test.sh [hunk 1]
```diff
@@ -48,6 +48,41 @@ if [[ "$SHARD_NUMBER" == "2" ]]; then
...
+if [[ "$SHARD_NUMBER" == "2" ]]; then
+  # Sanity check on torchbench w/ onnx
+  pip install pandas
+  log_folder="test/.torchbench_logs"
+  device="cpu"
+
+  mkdir -p "${log_folder}"
+  for mode in "accuracy" "performance"; do
+    for compiler in "dynamo-onnx" "torchscript-onnx"; do
+      for suite in "huggingface" "timm_models"; do
```
Comment (thiagocrepaldi) [inline scripts/onnx/test.sh:60]: use variables to hold `mode`, `compiler` and `suite` list values? makes it easier to extend later

**Review by thiagocrepaldi — COMMENTED** [review]
not blocking for me but @kit1980 and @malfet once advised us to move CI script logic to Python instead of shell to make it cross platform and less messy due to compatibility issues (bash vs zshell vs etc)

File: scripts/onnx/test.sh [hunk 2]
```diff
@@ -48,6 +48,44 @@ if [[ "$SHARD_NUMBER" == "2" ]]; then
   xdoctest torch.onnx --style=google --options="+IGNORE_WHITESPACE"
 fi
 
+if [[ "$SHARD_NUMBER" == "2" ]]; then
+  # Sanity check on torchbench w/ onnx
+  pip install pandas
+  log_folder="test/.torchbench_logs"
+  device="cpu"
```
Comment (thiagocrepaldi) [inline scripts/onnx/test.sh:55]: Do we still want `cpu` only as device or should we add a `for device in "$devices[@]}"; do`?

File: scripts/onnx/test.sh [hunk 3]
```diff
@@ -48,6 +48,44 @@ if [[ "$SHARD_NUMBER" == "2" ]]; then
...
+            "torchbench")
+                bench_args+=(-k resnet18)
+                ;;
+            "huggingface")
+                bench_args+=(-k ElectraForQuestionAnswering)
+                ;;
+            "timm_models")
+                bench_args+=(-k lcnet_050)
+                ;;
+            *)
```
Comment (thiagocrepaldi) [inline scripts/onnx/test.sh:78]: just confirming we don't have anything for "ado")

**Review by thiagocrepaldi — APPROVED** [review]
LGTM assuming we only care about cpu and no ADO suite

File: scripts/onnx/test.sh (hunk 2 above)
Comment (BowenBao) [reply scripts/onnx/test.sh:55]: cpu only since CI for us is cpu only

File: scripts/onnx/test.sh (hunk 3 above)
Comment (BowenBao) [reply scripts/onnx/test.sh:78]: no we don't, this is running in our github onnx ci.

**Pull request closed without merging**